app/form_persisters/form_persister.rb

```diff
@@ -25,37 +20,63 @@ def save_all(forms:)
     end
   end
 
-  private
+  def instance(form)
+    Instance.new(form: form, form_persister: self)
+  end
 
-    def before_save(form:)
-      create_files(form: form)
+  class Instance
+    attr_reader :form, :form_persister
+    delegate :adapter, :storage_adapter, to: :form_persister
+    delegate :persister, :query_service, to: :adapter
```

Comment:
I'm not a fan of delegating two levels deep (`self.persister` is analogue to `form_persister.adapter.persister`). I believe the given `form_persister` should expose the methods you want (and not rely on the form_perister exposing an adapter which in turn exposes a query_service and persister.